.github/workflows/ci.yml

```diff
@@ -40,4 +40,6 @@ jobs:
           target: ${{ matrix.target }}
           arch: x86_64
           profile: Nexus 6
-          script: ./gradlew connectedCheck
+          script: |
+            brew install parallel
+            parallel --retries 5 ::: "./gradlew connectedCheck"
```

Comment:
> 10 attempts was successful so retry never happened, anyway I propose to review and merge it

Shouldn't we instead fix the flaky tests?

Reply:
Ideally yes,

but AFAIK it is not always related to flaky tests, but mostly to flaky simulators)

For example:
 - https://github.com/hCaptcha/hcaptcha-android-sdk/runs/5676730153?check_suite_focus=true
 - https://github.com/hCaptcha/hcaptcha-android-sdk/runs/5535462245?check_suite_focus=true

Those failures happen only once and are never reproduced locally (at least on my end)

Anyway, I am ready to investigate such failures if we think that it makes sense.

As first step it will be useful to collect screenshots of UI on CI

P.S. This change affects only UI tests (i.e. `./gradlew connectedCheck`) for a regular unit tests (`./gradlew test`) we will not have any retry logic because they are executed on desktop JVM

Reply:
The simulator has animation disabled, maybe that can cause an issue on pipeline but works fine on local. 
I still believe the tests are flaky since it involves network calls, which can timeout for different reasons, which we may have not accounted for.

Isn't 5 too many retries? I would expect after 2 or 3 to just be a problem with the tests themselves.
Let's go with your retry option first and see if the tests persist to fail or not.